GVRf/Sample/gvr-vuforia/src/org/gearvrf/vuforiasample/VuforiaSampleScript.java

```diff
@@ -20,157 +13,157 @@
+
+        mMainScene.getMainCameraRig().getOwnerObject()
+                .addChildObject(mPassThroughObject);
+    }
+
+    private void createTeaPotObject() {
+        try {
+            teapot = new GVRSceneObject(mGVRContext,
+                    mGVRContext.loadMesh(new GVRAndroidResource(mGVRContext
+                            .getContext(), "teapot.obj")),
+                    mGVRContext.loadTexture(new GVRAndroidResource(mGVRContext
+                            .getContext(), "teapot_tex1.jpg")));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
 
-                    Matrix.multiplyMM(mConvertedMVMatrix, 0, convertMatrix, 0,
-                            mVuforiaMVMatrix, 0);
+        teapot.getTransform().setPosition(0f, 0f, -0.5f);
+        mMainScene.addSceneObject(teapot);
+    }
 
-                    float scaleFactor = ((ImageTarget) trackable).getSize()
-                            .getData()[0];
-                    Matrix.scaleM(mConvertedMVMatrix, 0, scaleFactor,
-                            scaleFactor, scaleFactor);
+    private float[] convertMatrix = { 1f, 0f, 0f, 0f, 0f, -1f, 0f, 0f, 0f, 0f,
+            -1f, 0f, 0f, 0f, 0f, 1f };
 
-                    mGVRMVMatrix = mGVRContext.getMainScene()
-                            .getMainCameraRig().getOwnerObject().getTransform()
-                            .getModelMatrix();
+    @Override
+    public void onStep() {
+        if (VuforiaSampleActivity.isVuforiaActive()) {
+            Renderer.getInstance().begin();
+            Renderer.getInstance().bindVideoBackground(0);
+            Renderer.getInstance().end();
+        }
+    }
 
-                    Matrix.multiplyMM(mTotalMVMatrix, 0, mGVRMVMatrix, 0,
-                            mConvertedMVMatrix, 0);
-                    bunny.getTransform().setModelMatrix(mTotalMVMatrix);
-                    break;
-                }
+    public void updateObjectPose(State state) {
+        // did we find any trackables this frame?
+        for (int tIdx = 0; tIdx < state.getNumTrackableResults(); tIdx++) {
+            TrackableResult result = state.getTrackableResult(tIdx);
+            Trackable trackable = result.getTrackable();
+            if (trackable.getId() == 1) {
+                Matrix44F modelViewMatrix_Vuforia = Tool
+                        .convertPose2GLMatrix(result.getPose());
+                mVuforiaMVMatrix = modelViewMatrix_Vuforia.getData();
+
+                Matrix.multiplyMM(mConvertedMVMatrix, 0, convertMatrix, 0,
+                        mVuforiaMVMatrix, 0);
+
+                float scaleFactor = ((ImageTarget) trackable).getSize()
+                        .getData()[0];
+                Matrix.scaleM(mConvertedMVMatrix, 0, scaleFactor, scaleFactor,
+                        scaleFactor);
+
+                mGVRMVMatrix = mGVRContext.getMainScene().getMainCameraRig()
+                        .getOwnerObject().getTransform().getModelMatrix();
```

Comment:
you can remove getOwnerObject() here.  just mGVRContext.getMainScene().getMainCameraRig().getTransform().getModelMatrix().